PowerPointLabs/PowerPointLabs/TooltipsLab/AttachTriggerAnimation.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Windows.Forms;
+using Microsoft.Office.Interop.PowerPoint;
+using PowerPointLabs.Models;
+using PowerPointLabs.TextCollection;
+
+namespace PowerPointLabs.TooltipsLab
+{
+    internal static class AssignTooltip
+    {
+        public static void AddTriggerAnimation(PowerPointSlide currentSlide, Selection selection)
+        {
+            ShapeRange selectedShapes = selection.ShapeRange;
+
+            if (selectedShapes.Count < 2)
+            {
+                MessageBox.Show(TooltipsLabText.ErrorLessThanTwoShapesSelected,
+                    TooltipsLabText.ErrorTooltipsDialogTitle);
+
+                // TODO: New Exception for TooltipsLab
+                throw new Exception();
+            }
+
+            Shape triggerShape = selectedShapes[1];
+
+            List<Shape> shapesToAnimate = GetShapesToAnimate(selectedShapes);
+
+            AddTriggerAnimation(currentSlide, triggerShape, shapesToAnimate);
+
+        }
+
+        public static void AddTriggerAnimation(PowerPointSlide currentSlide, Shape triggerShape, Shape calloutShape)
+        {
+            List<Shape> calloutShapeList = new List<Shape>();
+            calloutShapeList.Add(calloutShape);
+            AddTriggerAnimation(currentSlide, triggerShape, calloutShapeList);
+        }
+
+        private static void AddTriggerAnimation(PowerPointSlide currentSlide, Shape triggerShape, List<Shape> newShapesToAnimate)
+        {
+            TimeLine timeline = currentSlide.TimeLine;
+            // TODO: Get effect from settings
+            MsoAnimEffect fadeEffect = MsoAnimEffect.msoAnimEffectFade;
+
+            // Get the shapes that are already associated with trigger shape
+            List<Shape> shapesToAnimate = RemoveAnimationsInInteractiveSequence(currentSlide, triggerShape, newShapesToAnimate);
```

Comment:
The name "`RemoveAnimationsInInterativeSequence`" seems a bit misleading as it seems like its function is to GetShapes of some kind, but yet in the name there is no suggestion of getting shapes, sounds like it is just removing..?

Reply:
I understand your concern here, as much as I would like to make the function have a "single repsonsibility" it doesn't seem possible here. 

What do you suggest i rename the function, `GetShapesAndRemoveAnimationsInInteractiveSequence`?

Reply:
`GetShapesInInteractiveSequenceWithAnimationsRemoved` ?